signature/cose/envelope.go

```diff
@@ -382,22 +421,23 @@ func generateProtectedHeaders(req *signature.SignRequest, protected cose.Protect
 
 // generateUnprotectedHeaders creates Unprotected Headers of the COSE envelope
 // during Sign process.
-func generateUnprotectedHeaders(req *signature.SignRequest, unprotected cose.UnprotectedHeader) error {
+func generateUnprotectedHeaders(req *signature.SignRequest, signer cose.Signer, unprotected cose.UnprotectedHeader) {
 	// signing agent
 	unprotected[headerLabelSigningAgent] = req.SigningAgent
 
 	// certChain
-	certs, err := req.Signer.CertificateChain()
-	if err != nil {
-		return err
+	var certs []*x509.Certificate
+	switch s := signer.(type) {
+	case *remoteSigner:
```

Comment:
nit: `case *remoteSigner, *localSigner:`

Reply:
@binbin-li I actually have tried that but got the following error: s.certs undefined (type "github.com/veraison/go-cose".Signer has no field or method certs).
variable s becomes cose.Signer type, and it does not have certs field defined, so looks like I have to separate them into 2 cases.

Reply:
gotcha, good to know that.